module/extension/Director/tests/when_blocking.cpp

```diff
@@ -0,0 +1,91 @@
+#include <catch.hpp>
+#include <nuclear>
+
+#include "Director.hpp"
+#include "TestBase.hpp"
+
+// Anonymous namespace to avoid name collisions
+namespace {
+
+    struct SimpleTask {};
+
+    struct Condition {
+        enum Value { BLOCK, ALLOW } value;
+        Condition(const Value& v) : value(v) {}
+        operator int() const {
+            return value;
+        }
+    };
+
+    std::vector<std::string> events;
+
+    class TestReactor : public TestBase<TestReactor> {
+    public:
+        explicit TestReactor(std::unique_ptr<NUClear::Environment> environment)
+            : TestBase<TestReactor>(std::move(environment)) {
+
+            on<Provide<SimpleTask>, When<Condition, std::equal_to, Condition::ALLOW>>().then([this] {
+                // Task has been executed!
+                events.push_back("task executed");
+            });
+
+            /**************
+             * TEST STEPS *
+             **************/
+            on<Trigger<Step<1>>, Priority::LOW>().then([this] {
+                // At this point condition hasn't been emitted, so this task should be blocked by default
+                events.push_back("emitting task");
+                emit<Task>(std::make_unique<SimpleTask>());
+            });
+            on<Trigger<Step<2>>, Priority::LOW>().then([this] {
+                // Emitting a blocked condition
+                events.push_back("emitting blocked condition");
+                emit(std::make_unique<Condition>(Condition::BLOCK));
+            });
+            on<Trigger<Step<3>>, Priority::LOW>().then([this] {
+                // Emitting the task again to ensure it's not executed
+                events.push_back("emitting task again");
+                emit<Task>(std::make_unique<SimpleTask>());
+            });
+            on<Trigger<Step<4>>, Priority::LOW>().then([this] {
+                // This should make it run since it's now allowed
+                events.push_back("emitting allowed condition");
+                emit(std::make_unique<Condition>(Condition::ALLOW));
+            });
+            on<Trigger<Step<5>>, Priority::LOW>().then([this] {
+                // This task should run fine since it's allowed
+                events.push_back("emitting task again x2");
```

Comment:
```suggestion
                events.push_back("emitting task again _2");
```
`x2` implies that we emit the task twice here. I think you were trying to make the message unique and that `_2` would be better.

Reply:
True, I think `#2` helps even more signify it's number 2